query/physicalplan/binaryscalarexpr.go

```diff
@@ -217,3 +229,69 @@ func DictionaryArrayScalarEqual(left *array.Dictionary, right scalar.Scalar) (*B
 
 	return res, nil
 }
+
+func ArrayScalarContains(arr arrow.Array, right scalar.Scalar) (*Bitmap, error) {
+	res := NewBitmap()
+	switch a := arr.(type) {
+	case *array.Binary:
+		var r []byte
+		switch s := right.(type) {
+		case *scalar.Binary:
+			r = s.Data()
+		case *scalar.String:
+			r = s.Data()
+		}
+
+		for i := 0; i < a.Len(); i++ {
+			if a.IsNull(i) {
+				continue
+			}
+			if bytes.Contains(a.Value(i), r) {
+				res.Add(uint32(i))
+			}
+		}
+		return res, nil
+	}
+	return nil, fmt.Errorf("contains not implemented for %T", arr)
+}
+
+func DictionaryArrayScalarContains(left *array.Dictionary, right scalar.Scalar) (*Bitmap, error) {
+	res := NewBitmap()
+	var data []byte
+	switch r := right.(type) {
+	case *scalar.Binary:
+		data = r.Data()
+	case *scalar.String:
+		data = r.Data()
+	}
+
+	// This is a special case for where the left side should not equal NULL
+	if right == scalar.ScalarNull {
+		for i := 0; i < left.Len(); i++ {
+			if !left.IsNull(i) {
+				res.Add(uint32(i))
+			}
+		}
+		return res, nil
+	}
+
+	for i := 0; i < left.Len(); i++ {
+		if left.IsNull(i) {
+			continue
+		}
+
+		switch dict := left.Dictionary().(type) {
+		case *array.Binary:
+			if bytes.Contains(dict.Value(left.GetValueIndex(i)), data) {
+				res.Add(uint32(i))
+			}
+		case *array.String:
+			// TODO: Add unsafe type cast if necessary.
```

Comment:
Can we write a benchmark for this? I suspect this is already necessary right off the bat.